quinn-proto/src/connection.rs

```diff
@@ -1687,12 +1675,9 @@ where
                     self.path_challenge = None;
                 }
                 Frame::MaxData(bytes) => {
-                    let was_blocked = self.flow_blocked();
-                    self.max_data = cmp::max(bytes, self.max_data);
-                    if was_blocked && !self.flow_blocked() {
-                        for stream in self.blocked_streams.drain() {
-                            self.events.push_back(Event::StreamWritable { stream });
-                        }
+                    self.streams.max_data(bytes);
+                    for stream in self.streams.drain_unblocked() {
```

Comment:
This loop seems to appear in a few places. Can we do it in one place (maybe inside `poll()`?). (Could be a separate commit if you prefer.)

Reply:
Great catch! That makes this more readable *and* more efficient.